contracts/exchangeIssuance/FlashMintWrapped.sol

```diff
@@ -17,220 +17,1087 @@
+    return
+      _redeemExactSet(
+        _setToken,
+        _outputToken,
+        _amountSetToken,
+        _minOutputReceive,
+        _swapData,
+        _invokeUnwrapData,
+        false
+      );
+  }
+
+  /**
+   * Redeems an exact amount of SetTokens for ETH.
+   * The sender must have approved the _amountSetToken of _setToken to this contract.
+   *
+   * @param _setToken              Address of the SetToken to be redemeed
+   * @param _amountSetToken        Amount of SetTokens to redeem
+   * @param _minOutputReceive      Minimum amount of output tokens to be received
+   * @param _swapData              ComponentSwapData (underlyingERC20 -> output token) for each _redeemComponents in the same order
+   * @param _invokeUnwrapData      ComponentInvokeWrapData (wrapped Set component -> underlyingERC20) for each _redeemComponents in the same order
+   *
+   * @return outputAmount          Amount of ETH sent to the caller
+   */
+  function redeemExactSetForETH(
+    ISetToken _setToken,
+    uint256 _amountSetToken,
+    uint256 _minOutputReceive,
+    ComponentSwapData[] calldata _swapData,
+    ComponentInvokeWrapData[] calldata _invokeUnwrapData
+  ) external nonReentrant returns (uint256) {
+    // output token for all operations is WETH (it will be unwrapped in the end and sent as ETH)
+    IERC20 outputToken = IERC20(dexAdapter.weth);
+
+    return
+      _redeemExactSet(
+        _setToken,
+        outputToken,
+        _amountSetToken,
+        _minOutputReceive,
+        _swapData,
+        _invokeUnwrapData,
+        true
+      );
+  }
+
+  /**
+  * Gets the wrap callData required for issuance for the passed in ComponentWrapData
+  *
+  * @param _wrapModule                  Address of wrapModule that the integrations in ComponentWrapData are assigned to
+  * @param _wrapData                    ComponentWrapData (underlyingERC20 -> wrapped Set component) for each requiredComponents in the same order
+  *
+  * @return ComponentInvokeWrapData[]   data used for input in issuance methods
+  */
+  function getWrapCallData(
+    address _wrapModule, 
+    ComponentWrapData[] calldata _wrapData
+  ) external view returns(ComponentInvokeWrapData[] memory){
+
+    ComponentInvokeWrapData[] memory invokeWrapData = new ComponentInvokeWrapData[](_wrapData.length);
+
+    for (uint256 i = 0; i < _wrapData.length; ++i) {
+
+      invokeWrapData[i] = ComponentInvokeWrapData({
+        callTarget: address(0),
+        value: 0,
+        callData: ""
+      });
+
+
+      if(_wrapData[i].fromToken == _wrapData[i].toToken) {
+        // no wrapping needed
+        continue;
+      }
+
+      // get the wrap adapter directly from the integration registry
+      IWrapV2Adapter wrapAdapter = IWrapV2Adapter(_getAndValidateAdapter(_wrapData[i].integrationName, _wrapModule));
+
+      // get wrap call info from adapter
+      (invokeWrapData[i].callTarget, invokeWrapData[i].value, invokeWrapData[i].callData) = wrapAdapter
+        .getWrapCallData(
```

Comment:
I think it'd be better to integrate this call into the issuance / redemption methods, since it would avoid many of the issues I mentioned elsewhere.

Reply:
Implemented as discussed